File: algo-expert/arrays/easy questions/two_number_sum.py
# def two_number_sum(array, targetSum):
#     if not array:
#         return []
#     for i in range(0, len(array) - 1):
#         for j in range(i + 1, len(array)):
#             if array[j] + array[i] == targetSum:
#                 return [array[i], array[j]]


def twoNumberSum(array, targetSum):
    for num in array:
        target = targetSum - num
        if target in array and target != num:
            return [num, target]

    return []


print(twoNumberSum([3, 5, -4, 8, 11, 1, -1, 6], 10))

File: algoexpert/arrays/easy_questions/twoNumberSum.js
function twoNumberSum(array, targetSum) {
    // Write your code here.
    for (let i = 0; i < array.length - 1; i++) {
      let x = array[i]
      if (x = targetSum)
      for (let j = 0; j < array.length; j++) {
        let y = array[j + 1 + i]
        if (x + y == targetSum) {
          console.log(`[${x}, ${y}]`)
        }
        
      }
    }
  }

  twoNumberSum([15], 15)